### DISCUSSION TITLE
Latest Release Labeling (and on Front Page)

### DISCUSSION THREAD

**Discussion by mathomp4** [community/community#6389] [Other Feature Feedback, Questions, & Ideas] [answered]
I am a big big fan of the new Releases Page and I just discovered a nice thing today which is that I had to release a version based on an old minor version today and I saw that [on my Releases page](https://github.com/GEOS-ESM/MAPL/releases) the actual latest Semver release is on top:

<img width="1291" alt="Screen Shot 2021-10-15 at 2 23 10 PM" src="https://user-images.githubusercontent.com/2982494/137534765-faba7710-ff90-448d-b955-bcef5f16fc22.png">

Huzzah! 🎉 

That said, I'd prefer the green "Latest" label/icon to actually point to the latest Semver and not the latest-in-date release.

Also it would be really nice if the [front page of the repo](https://github.com/GEOS-ESM/MAPL) would also reflect that. Unfortunately, it seems to be going by date rather than by Semver as well:

<img width="360" alt="Screen Shot 2021-10-15 at 2 25 36 PM" src="https://user-images.githubusercontent.com/2982494/137535074-c02119c6-206d-418c-88d6-3b9ec6cc83ca.png">

I can understand why GitHub does this as chronological order is a valid sort. Maybe if when we make a release, we could have a third checkbox question a la:

<img width="804" alt="Screen Shot 2021-10-15 at 2 26 44 PM" src="https://user-images.githubusercontent.com/2982494/137535206-f4b3dc20-11fc-4c4a-ae91-2b74bfe264c6.png">

but where it says:

- [x] **Mark this as latest release**

and by default it could be checked, but if we release a bugfix on an old branch, we could unselect that box?

**Comment by MylesBorins** [comment]
Hey @mathomp4 thanks for opening this discussion!

Definitely with you that we need to rethink / improve the behavior of "latest"... especially based on how we handle the sort for releases made on the same day. In fact we were going to have the latest release show full bleed rather than abridged but dropped that from a set of changes this week because the latest release currently is not consistently at the top of the page.

Working through the behavior of latest and coming up with something more reliable / consistent is something we have queued up to tackle next sprint... but we are still not 100% exactly how it will land.

I'll keep you up to date as we progress and reference this discussion as we scope it!

**Reply by mathomp4** [reply]
@MylesBorins Thanks!

**Comment by TheMapotakes** [comment]
Wee!

**Comment by MylesBorins** [comment, marked as answer]
We just shipped a fix to how we do latest, PTAL!

**Reply by mathomp4** [reply]
Oooh. Looking good! I'll do some more testing tomorrow!

**Reply by mathomp4** [reply]
@MylesBorins Yeah. I think this got broken again. On my project:

https://github.com/GEOS-ESM/MAPL/releases

You'll see it thinks 2.12.2 is "newer" than 2.18.3

**Reply by pablonete** [reply]
Fair feedback. The reason is that GitHub sort releases by date (without time), then on the same day, it applies semver. So 2.12.2 is newer than 2.18.3 because it was published at a recent calendar date. If both where published on the same day, 2.18.3 would be latest, no matter the relative order.

**Reply by mathomp4** [reply]
Ahh. I guess it would be nice to consider semver first, then date. I mean, unless your releases are tagged by date.

I really wish there was some way I could "manually" mark a release as the latest when I make it. Then GitHub wouldn't need to care. I say release v1.2.3 is latest, it doesn't matter if the next day I release v5.6.7.

**Comment by mathomp4** [comment]
@MylesBorins I think this might have broken again. 😦 

I just released a v2.8.0.1 (don't ask) on a repo whose actual "latest" release is v2.15.0. But when you go to the site:

https://github.com/GEOS-ESM/MAPL/
https://github.com/GEOS-ESM/MAPL/releases

it says that 2.8.0.1 is the "latest" release when it only is in time, not SemVer.
